### ISSUE TITLE
Dangling completion pop-up windows

### ISSUE THREAD

**Issue by Frederick888** [neoclide/coc.nvim#2444]
## Result from CocInfo

```
## versions

vim version: NVIM v0.4.4
node version: v14.13.1
coc.nvim version: 0.0.79-8da7b2139d
coc.nvim directory: /Users/fzhang/.vim/plugged/coc.nvim
term: screen-256color
platform: darwin

## Output channel: explorer


## Output channel: git

Looking for git in: git
> git rev-parse --show-toplevel
resolved root of /Users/fzhang/Programming/Java/eclipse.jdt.ls/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/Environment.java: /Users/fzhang/Programming/Java/eclipse.jdt.ls
> git ls-files -- org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/Environment.java
> git symbolic-ref --short HEAD
> git diff --name-status
> git diff --staged --name-status
> git ls-files --others --exclude-standard
> git --no-pager show :org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/Environment.java

## Output channel: highlight

[Info  - 12:28:43 pm] Highlight server running in node v14.13.1

## Output channel: snippets

[Info 12:28:43 pm] Using ultisnips directories: UltiSnips /Users/fzhang/.config/coc/ultisnips
[Info 12:28:43 pm] Using ultisnips python command: pyx
```

## Describe the bug

When working on large projects, if I type relatively fast, the completion window is not automatically closed from time to time.

~~And not sure whether it's relevant, when I was trying to reproduce this problem to get a screenshot, the dangling pop-up were all for yanked texts.~~ Just happened with `coc-yank` disabled, but now it definitely happens less often.

## Reproduce the bug

**We will close your issue when you don't provide minimal vimrc and we can't
reproduce it**

- Clone https://github.com/eclipse/eclipse.jdt.ls as an example

- Install `coc-java` `coc-yank`

- Create file `mini.vim` with：

  ```vim
  set nocompatible
  set runtimepath^=/path/to/coc.nvim
  filetype plugin indent on
  syntax on
  set hidden
  ```

- Start (neo)vim with command: `vim -u init.vim org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/Environment.java`

- Operate vim.

Full list of loaded (`*` symbol) extensions:

- coc-yank
- coc-todolist
- coc-snippets
- coc-lists
- coc-java-debug
- coc-java
- coc-highlight
- coc-git
- coc-explorer
- coc-eslint
- coc-diagnostic

## Screenshots (optional)

<img width="1299" alt="Screen Shot 2020-10-13 at 12 27 44 pm" src="https://user-images.githubusercontent.com/4507647/95805011-1113ff00-0d50-11eb-9686-f42eb12610c4.png">
The text in the shown pop-up were yanked from another project.

**Comment by Frederick888** [comment]
This is happening again using 63cbd7bfecb6ed5598b136e859c675ae8e7156ea: 
<img width="689" alt="image" src="https://user-images.githubusercontent.com/4507647/102180032-509bca80-3efc-11eb-8524-b451cf6c6637.png">

vimrc used to get the screenshot above:
```vim
set runtimepath^=~/.vim runtimepath+=~/.vim/after
let &packpath = &runtimepath

" vim-plug
call plug#begin()

Plug 'neoclide/coc.nvim', {'branch': 'release'}

call plug#end()

" ==== Start Coc Configurations ====

" Some servers have issues with backup files, see #649
set nobackup
set nowritebackup

" Better display for messages
set cmdheight=2

" You will have bad experience for diagnostic messages when it's default 4000.
set updatetime=300

" Don't give |ins-completion-menu| messages.
set shortmess+=c

" Always show the signcolumn, otherwise it would shift the text each time
" diagnostics appear/become resolved.
if has("patch-8.1.1564")
  " Recently vim can merge signcolumn and number column into one
  set signcolumn=number
else
  set signcolumn=yes
endi

" Single column signs
let g:coc_status_error_sign = "\U2717 "
let g:coc_status_warning_sign = "\U26A0 "

" Remap keys for gotos
nmap <silent> gd <Plug>(coc-definition)
nmap <silent> gy <Plug>(coc-type-definition)
nmap <silent> gi <Plug>(coc-implementation)
nmap <silent> gr <Plug>(coc-references)

" ==== End Coc Configurations ====

" vim: set tabstop=4 softtabstop=0 expandtab shiftwidth=2 smarttab:
```

In the screenshot, contents of the dangling pop-up is from `coc-yank`, though after using it for a whole day I've seen this issue being caused by other sources as well, e.g. completion, snippets, etc.

**Comment by chemzqm** [comment]
Should be fixed by 342d677c

**Comment by Frederick888** [comment]
@chemzqm I've been using this since yesterday afternoon and now dangling pop-ups don't even happen in the first place! Thanks a ton.
